app/src/main/java/com/example/stretchy/activity/MainActivity.kt

```diff
@@ -24,47 +19,53 @@ class MainActivity : ComponentActivity() {
         super.onCreate(savedInstanceState)
         activityComponent = ActivityComponent.create(this)
         setContent {
+            var showExportPermissionDialog by remember { mutableStateOf(false) }
+            var showImportPermissionDialog by remember { mutableStateOf(false) }
             Navigation(
                 activityComponent,
-                {
-                    grantPermissions(WRITE_EXTERNAL_STORAGE)
-                },
-                {
-                    grantPermissions(READ_EXTERNAL_STORAGE)
-                }
+                onExportClick = { showExportPermissionDialog = true },
+                onImportClick = { showImportPermissionDialog = true }
             )
+            if (showExportPermissionDialog) {
```

Comment:
why is this whole code held in activity? Activity should be as simple as it can be, those click listeners whould not exist if this code is held inside Navigation(...) composable func.